docs/designs/api-design.md

```diff
@@ -0,0 +1,343 @@
+# API Design
+
+The following document describes the individual APIs used in this application, request/response models,
+and HTTP Status Codes. In the future, this document will be superseeded by the Swagger model definition as the
+source of truth. At that point, we will use this document the behavior of individual APIs instead.
+
+Refer to the [Data Model Doc](./data-models.md) for source of truth details about each entity
+referrred in the doc here.
+
+## NOTE
+
+- All the APIs use the following prefix: `api.example.com/v1/`. The prefix is skipped in the following descriptions for simplicity.
+
+- Authentication parameters is not shown the following API definitions for simplicity
+
+- Common HTTP Status like `401 Unauthorized` or `500 _internal Server Error` are not shown here for simplicity
+
+- We try to use standard REST HTTP verbs as much as possible. Any deviations are described clearly in the API.
+
+- For simplicity, we are creating APIs around the business logic and not the underlying data models.
+  This is a deviation from best-practices. But it keeps the frontend logic fairly simple. In the backend
+  implementation, we will have high-level business logic methods that map to lower level data-model specific methods
+  to interface with the database
+
+- As much as possiible, we have tried to separate an object's data from the way it is visually
+  presented on the screen. This should simplify the backend logic and hopefully keep the
+  data storage de-coupled from the visual presentation. This should also allow the UI to be
+  rendered quickly without waiting for the entire data model to be parsed & processed.
+
+- All dates returned by the API are in UTC. UI is responsible for converting to local
+  timezone
+
+- All keys in the API request/response use Camel Case starting with a lower case letter.
+
+- In response JSONs, the core data elements are enclosed in a top-level key
+  representing the entity returned by the response. Example: In the `Get Programs API`, all
+  contents of the Program entity are stored within a top-level key called "program" -
+  `{"program": {...}}`. This is a more extensible design allowing us to send other metadata
+  along with the core entity.
+
+- All APIs have an `_internal` blob containing metadata used for backend's house keeping purposes.
+
+## User
+
+Get details about the logged in user
+
+`GET /user`
```

Comment:
I would suggest to name this endpoint `/me` instead `/user`, we can use `/user` path later for administrator API  to manage users.

Reply:
Not opposed to /me instead of /user but any future admin API should be implemented separately (isolated)